src/Simplex_tree/include/gudhi/Simplex_tree.h

```diff
@@ -1030,19 +1087,41 @@ class Simplex_tree {
    */
 
   Cofaces_simplex_range cofaces_simplex_range(const Simplex_handle simplex, int codimension) {
-    Cofaces_simplex_range cofaces;
     // codimension must be positive or null integer
     assert(codimension >= 0);
-    Simplex_vertex_range rg = simplex_vertex_range(simplex);
-    std::vector<Vertex_handle> copy(rg.begin(), rg.end());
-    if (codimension + static_cast<int>(copy.size()) > dimension_ + 1 ||
-        (codimension == 0 && static_cast<int>(copy.size()) > dimension_))  // n+codimension greater than dimension_
+
+    if constexpr (Options::link_nodes_by_label) {
+      // faster cofaces computation
+      if (codimension == 0) {
+        // Always true for a star
+        select = [](Const_boost_iterator) { return true; };
+      } else {
+        // Specific coface case
+        select = [this, simplex, codimension](Const_boost_iterator iter) {
+            return ((this->dimension(simplex) + codimension) == this->dimension(iter));
```

Comment:
```suggestion
        select = [this, dim=this->dimension(simplex)+codimension](Const_boost_iterator iter) {
            return dim == this->dimension(iter);
```
Or something like that to avoid recomputing the dimension of `simplex` all the time?

Reply:
I am ok with your proposal, done on 2adeccc